.github/workflows/deploy_docs_and_viewer.yml

```diff
@@ -0,0 +1,21 @@
+name: deploy docs and viewer
+on:
+    workflow_dispatch:
+#   push:
+#     branches:
+#       - master
+#       - main
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.x
+    #   - run: cd docs
+    #   - run: pwd
+      - run: pip install mkdocs-material
+      - run: git submodule update --init --recursive
+    #   &?
+      - run: mkdocs gh-deploy --config-file docs/mkdocs.yml --force --site-dir docs/site
```

Comment:
I am still not quite sure this will work. But we can iterate on this once we try to deploy it.